src/Plugin/OgGroupResolver/Entity.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Drupal\og\Plugin\OgGroupResolver;
+
+use Drupal\Core\Entity\ContentEntityBase;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\og\GroupTypeManager;
+use Drupal\og\Og;
+use Drupal\og\OgGroupResolverBase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Get the group from the given viewed entity.
+ *
+ * @OgGroupResolver(
+ *  id = "entity",
+ *  label = "Entity",
+ *  description = @Translation("Get the group from the current entity, by checking if it is a group or a group content entity.")
+ * )
+ */
+class Entity extends OgGroupResolverBase {
+
+  /**
+   * The route match service.
+   *
+   * @var RouteMatchInterface
+   */
+  protected $routeMatch;
+
+  /**
+   * The group type manager service.
+   *
+   * @var \Drupal\og\GroupTypeManager
+   */
+  protected $groupTypeManager;
+
+  /**
+   * Constructs a Drupal\Component\Plugin\PluginBase object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Routing\RouteMatchInterface $route_match
+   *   The route match service.
+   * @param \Drupal\og\GroupTypeManager $group_type_manager
+   *   The group type manager service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, RouteMatchInterface $route_match, GroupTypeManager $group_type_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->routeMatch = $route_match;
+    $this->groupTypeManager = $group_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('current_route_match'),
+      $container->get('og.group_type_manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getGroup() {
+    foreach ($this->routeMatch->getParameters() as $parameter) {
+      if (!($parameter instanceof ContentEntityBase)) {
+        continue;
+      }
+
+      if ($this->groupTypeManager->isGroup($parameter->getEntityTypeId(), $parameter->bundle())) {
+        return $parameter;
```

Comment:
The interface states that ::getGroups() should return EntityInterface[] but here returns a single item. Should be `return [$parameter];`?